File: Backend/commit-reader/src/main/java/com/fleetstudio/commit/reader/model/GitHubCommitModel.java
package com.fleetstudio.commit.reader.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.util.List;

@Data
public class GitHubCommitModel {

    @JsonProperty("sha")
    String commitSha;

    @JsonProperty("commit")
    CommitModel commit;

    @JsonProperty("parents")
    List<ParentModel> parents;

    @JsonProperty("files")
    List<FilesModel> files;
}

File: Frontend/git-commit-reader-frontend/src/model/git-hub-commit-model.ts
import { CommitModel } from "./commit-model";
import { FileModel } from "./file-model";
import { ParentModel } from "./parent-model";

export class GitHubCommitModel {
    sha: string = '';
    commit: CommitModel = new CommitModel();
    parents: ParentModel[] = [];
    files: FileModel[] = [];
}